enhancements/cluster-logging/performance-tuning.md

```diff
@@ -0,0 +1,241 @@
+---
+title: performance-tuning
+authors:
+  - "@alanconway"
+reviewers:
+  - "@jcantrill"
+approvers:
+  - "@jcantrill"
+api-approvers:
+  - "@jcantrill"
+creation-date: 2023-12-15
+last-updated: 2023-12-15
+tracking-link:
+  - https://issues.redhat.com/browse/OBSDA-549
+see-also:
+replaces:
+superseded-by:
+---
+
+# Performance tuning
+
+## Summary
+
+A _perfance tuning_ API to control performance, reliability and special protocol features of an output,
+without exposing the complexity of the underlying collector configuration.
+
+**Note**
+- Only vector will be supported, there are no plans to back-port to fluentd.
+- Existing `output[].limits` allows rate limiting, that is separate from this proposal.
+
+## Motivation
+
+Performance tuning in the underlying collector configuration is complex and error prone.
+Achieving a performance goal may involve configuring multiple parts of the delivery pipeline consistently.
+This would be difficult and error prone for users, but it can be safely automated by our configuration generator.
+
+### User Stories
+#### As a cluster logging administrator I want to minimize log loss.
+
+``` yaml
+outputs:
+ new - name: minimise-log-loss
+   tuning:
+     delivery: safe
+```
+
+#### As a cluster logging administrator I want to maximize throughput.
+
+``` yaml
+outputs:
+  - name: maximise-throughput
+    tuning:
+      delivery: fast
+```
+
+#### As a cluster logging administrator I want to tweak features of a specific output.
+
+``` yaml
+outputs:
+  - name: detailed-tweaks
+    tuning:
```

Comment:
@cahartma transfering our discussion here:
 you mentioned in scrum that you were not fond of 'fast' and 'safe'.  Maybe we should consider an alternate to identify the expectation.  We are trying to make  an optimization between throughput and durability.  Maybe thats the proper identifier.  The intent I believe is to provide some "sane" defaults though without testing we really don't know what that is.  it could even differ based on output.  This means I think we need a third option IMO which is... tweak yourself.  so maybe 
```
tuning:
  limits:
     maxRecords: #int
  optimization:
    type: throughput|durability|manual #durable?
    batch:
      maxEvents:
      maxBytes:  #or some equivalent we could calc from something like 500Mb
    buffer:
      whenFull:  #block|dropNewest  or some equivalent
    request:
      maxDelay
    compression:   
```

Reply:
Switched to standard messaging AtMostOnce/AtLeastOnce terminology. Not hugely intuitive outside the messaging world, but easy to explain and familiar to anyone who has reliable messaging experience.
I feel this is much better than fast/safe in many ways, thanks for pushing.